src/ghostfragment/fragmenting/inter_finder.cpp

```diff
@@ -1,55 +1,81 @@
 #include "inter_finder.hpp"
-#include <iostream>
 #include <algorithm>
+#include <chemist/nucleus/fragmented_nuclei.hpp>
 #include <cmath>
 #include <cstdlib>
+#include <iostream>
 
 using namespace ghostfragment::fragmenting;
 
+using nuke_type = chemist::FragmentedNuclei;
+
 using frag_type   = std::vector<std::size_t>;
 using system_type = std::vector<frag_type>;
 
 using group_type = std::vector<std::size_t>;
 
 using log_type = std::map<std::pair<std::size_t, std::size_t>, inter_type>;
 
-void find_group_intersections(
-  const system_type& fragments, inter_type inter, std::size_t recent_add,
-  res_type& m, log_type& log, group_type& group) {
+res_type find_group_intersections(const system_type& fragments,
+                                  inter_type inter, std::size_t recent_add,
+                                  res_type m, const log_type& log,
+                                  group_type& group) {
     if(group.size() > 2) {
         for(std::size_t i = 0; i < group.size() - 1; ++i) {
-            if(log.count(std::make_pair(group[i], recent_add)) == 0) return;
+            if(log.count(std::make_pair(group[i], recent_add)) == 0) return m;
         }
         inter_type::iterator it = set_intersection(
           inter.begin(), inter.end(), fragments[recent_add].begin(),
           fragments[recent_add].end(), inter.begin());
         inter.resize(it - inter.begin());
-    } else if(group.size() == 2)
-        inter = log[std::make_pair(group[0], group[1])];
+    } else if(group.size() == 2) {
+        std::pair<int, int> p(group[0], group[1]);
+        try {
+            inter = log.at(p);
+        } catch(...) { inter.clear(); }
```

Comment:
Generally speaking you should never need `try`/`catch` blocks in C++. 
```suggestion
        if(log.count(p))
            inter = log.at(p);
        else
         inter.clear();
```

Reply:
Forgot I could use count to check if it existed first. Thank you.